src/main/java/org/cloudfoundry/promregator/auth/CertificateBasedTokenFetcher.java

```diff
@@ -0,0 +1,55 @@
+package org.cloudfoundry.promregator.auth;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.net.ssl.SSLContext;
+
+import org.apache.http.NameValuePair;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.message.BasicNameValuePair;
+import org.cloudfoundry.promregator.auth.OAuth2XSUAAEnricher.TokenResponse;
+import org.cloudfoundry.promregator.config.OAuth2XSUAAAuthenticationConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sap.cloud.security.mtls.SSLContextFactory;
+
+public class CertificateBasedTokenFetcher extends TokenFetcher {
```

Comment:
The name of the class looks a little too generic to me. What about `CertificateBasedAuth2JWTFetcher`?

Reply:
Obsolet. The class does not exist anymore.